File: WSS_12_04-2023/WSS/Dispatch.vb
Module Dispatch
    Public TagsList As New Tags
    Public Tag As New Tag
    Public Sub RND_Value()
        For i As Short = 0 To TagsList.Count - 1
            TagsList(i).Value = Val(TagsList(i).Value) + Rnd(1)
        Next
    End Sub
    Public Sub init_Core(max_index As Short)
        'Dim ls As List(Of Integer)
        For i As Integer = 0 To max_index
            Dim Tag As New Tag
            Tag.ID = i ' Lias ID need to mutch with Tag.ID 
            Tag.ID_Name = "tag_id_" & i
            Tag.field_Name = "tag_name_id_" & i
            Tag.Scale = 3
            Tag.Value = Rnd(1) * i
            'Tag.c_Date = Now
            'ls = New List(Of Integer)
            'ls.Add(i)
            'Tag.SubsADD = i
            TagsList.Add(Tag)
        Next
        'For Each i In TagsList(5).Subs
        ' Dim f = i
        'Next
    End Sub

End Module

File: WpfWSS_CS/Dispatch.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace WpfWSS_CS
{
    public class Dispatch
    {
        public static Tags TagsList = new Tags();
        public static Tag Tag = new Tag();

        public static void RND_Value()
        {
            for (short i = 0; i < TagsList.Count; i++)
            {
                TagsList[i].Value = Convert.ToDouble(TagsList[i].Value) + new Random().NextDouble();
            }
        }

        public static void init_Core(short max_index)
        {
            for (int i = 0; i <= max_index; i++)
            {
                Tag tag = new Tag();
                tag.ID = i;
                tag.ID_Name = "tag_id_" + i;
                tag.field_Name = "tag_name_id_" + i;
                tag.Scale = 3;
                tag.Value = new Random().NextDouble() * i;
                TagsList.Add(tag);
            }
        }
    }

}
